Core/Utilities/SiteCEMapping.py

```diff
@@ -49,35 +73,29 @@ def getSiteCEMapping( gridName = '' ):
   return S_OK( siteCEMapping )
 
 #############################################################################
-def getCESiteMapping( gridName = '' ):
+def getCESiteMapping( gridName = None ):
   """ Returns a dictionary of all CEs and their associated site, e.g.
       {'ce101.cern.ch':'LCG.CERN.ch', ...]}
       Assumes CS structure of: /Resources/Sites/<GRIDNAME>/<SITENAME>
   """
   ceSiteMapping = {}
-  gridTypes = gConfig.getSections( '/Resources/Sites/', [] )
+  gridTypes = __getGridTypes( gridName )
   if not gridTypes['OK']:
-    gLogger.warn( 'Problem retrieving sections in /Resources/Sites' )
     return gridTypes
-
   gridTypes = gridTypes['Value']
-  if gridName:
-    if not gridName in gridTypes:
-      return S_ERROR( 'Could not get sections for /Resources/Sites/%s' % gridName )
-    gridTypes = [gridName]
 
   for grid in gridTypes:
     sites = gConfig.getSections( '/Resources/Sites/%s' % grid, [] )
-    if not sites['OK']: #gConfig returns S_ERROR for empty sections until version
-      gLogger.warn( 'Problem retrieving /Resources/Sites/%s section' % grid )
+    if not sites['OK']:  # gConfig returns S_ERROR for empty sections until version
+      gLogger.error( 'Problem retrieving /Resources/Sites/%s section' % grid, sites['Message'] )
       return sites
```

Comment:
because it is inside a loop... I can extract the value of grid from the name... will look into it